### ISSUE TITLE
Training results with RT-DETR are not reproducible using the same arguments

### ISSUE THREAD

**Issue by aitormfotos** [ultralytics/ultralytics#17558]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I trained an RT-DETR-l model twice using the same .yaml configuration file with identical training arguments. Despite using the same data, a fixed seed, and disabling random augmentations, the results are not reproducible between runs. The metrics and loss values diverge across trainings.

It is worth noting that this issue does not occur with other YOLO models (e.g., YOLOv8), where identical arguments and configurations produce reproducible results across multiple runs.

Training Code
Here is the training code used:
![train](https://github.com/user-attachments/assets/735ef123-e9a4-45a8-a475-43f9208907af)

.yaml Configuration File
Below is the .yaml file used for both training runs:
![train](https://github.com/user-attachments/assets/0465a95c-af70-4841-a269-5d49d792bf40)

### Environment

Ultralytics 8.3.31 🚀 Python-3.10.14 torch-2.4.1+cu121 CUDA:0 (NVIDIA GeForce GTX 1060 6GB, 6057MiB)
Setup complete ✅ (16 CPUs, 31.1 GB RAM, 218.1/232.0 GB disk)

OS                  Linux-5.14.0-1055-oem-x86_64-with-glibc2.31
Environment         Linux
Python              3.10.14
Install             pip
RAM                 31.10 GB
Disk                218.1/232.0 GB
CPU                 Intel Core(TM) i7-10700K 3.80GHz
CPU count           16
GPU                 NVIDIA GeForce GTX 1060 6GB, 6057MiB
GPU count           1
CUDA                12.1

numpy               ✅ 2.1.3>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.1+cu121>=1.8.0
torchvision         ✅ 0.19.1+cu121>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.11>=2.0.0
numpy               ✅ 2.1.3<2.0.0; sys_platform == "darwin"
torch               ✅ 2.4.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

![train](https://github.com/user-attachments/assets/78f30e01-91dc-43de-aea1-0fb2af2ec78b)
![args](https://github.com/user-attachments/assets/0df13744-fa12-4703-8ff8-42969c80bd20)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @aitormfotos, thank you for bringing this to our attention! 🚀 This is an automated response, and a dedicated Ultralytics engineer will also assist with your issue soon.

In the meantime, here are some steps you can follow to help us diagnose the problem:

If this is a 🐛 Bug Report, please ensure that you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). It's key for us to analyze the situation accurately.

For more insights, you might want to visit our [Docs](https://docs.ultralytics.com) where you can explore various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that may already address your question.

If this pertains to training models, do have a look at our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure all configurations are optimized.

Stay connected with the Ultralytics community for real-time support and discussions. Join our [Discord](https://ultralytics.com/discord) 🎧 for chat, [Discourse](https://community.ultralytics.com) for discussions, or our [Subreddit](https://reddit.com/r/ultralytics) for sharing insights.

## Upgrade

Make sure you are using the latest `ultralytics` version, along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). The latest updates might have addressed this issue:

```bash
pip install -U ultralytics
```

## Environments

Experience YOLO with these verified environments:

- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** with [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** with [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image** with [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Check our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) badge for the latest CI test results. Ensure your issue isn't already resolved in the current operations.

We appreciate your patience and understanding as we work together to resolve this issue! 😊

**Comment by Y-T-G** [comment]
RT-DETR training isn't deterministic

https://github.com/ultralytics/ultralytics/blob/4453ddab93c390c11c9f8bb8d90a761a4cec3342/ultralytics/models/rtdetr/train.py#L21
